test-alerts.cpp

```diff
@@ -4,5 +4,79 @@
 #include "typewise-alert.h"
 
 TEST_CASE("infers the breach according to limits") {
-  REQUIRE(inferBreach(12, 20, 30) == TOO_LOW);
+  REQUIRE(inferBreach(-12, 0, 35) == TOO_LOW);
+  REQUIRE(inferBreach(101,100,105) == NORMAL);
+  REQUIRE(inferBreach(25,20,30)==NORMAL);
+  REQUIRE(inferBreach(0, 0, 35) == TOO_LOW);
+  REQUIRE(inferBreach(105,100,105) == TOO_HIGH);
+  REQUIRE(inferBreach(120,100,105) == TOO_HIGH);
+}
+
+TEST_CASE("Check for Upper limit check functionality") {
+  REQUIRE(checkValueInUpperLimit(100, 99) == 1);
+  REQUIRE(checkValueInUpperLimit(100,100) == 1);
+  REQUIRE(checkValueInUpperLimit(-98,100) == 0);
+}
+
+TEST_CASE("Check for lower limit check functionality") {
+  REQUIRE(checkValueInLowerLimit(10, 0) == 0);
+  REQUIRE(checkValueInLowerLimit(10,-100) == 0);
+  REQUIRE(checkValueInLowerLimit(-98,100) == 1);
+  REQUIRE(checkValueInLowerLimit(-102,-100) == 1);
+}
+
+
+TEST_CASE("Check the functionality of mail notification")
+{
+  REQUIRE(sendToEmail(NORMAL)==E_NOT_OK);
```

Comment:
These tests dont check the content of the email. Consequence is that the email and controller messages need to be tested with real email and controllers being present.
Try dependency injection to make an interface for sending email, implementing the stub to store the contents and then asserting on it.